collectors/metrics/pkg/metricfamily/label.go

```diff
@@ -53,21 +53,44 @@ func appendLabels(
 	existing []*clientmodel.LabelPair,
 	overrides map[string]*clientmodel.LabelPair) []*clientmodel.LabelPair {
 	var found []string
+
+	// override matching existing labels
 	for i, pair := range existing {
 		name := pair.GetName()
+		// remove any name == "" or value == nil
+		if name == "" || pair != nil {
```

Comment:
should we use pair.getValue()?

Reply:
Oh, I totally misunderstood what we were checking for. Updating to your suggestion